internal/services/containers/kubernetes_flux_configuration_resource.go

```diff
@@ -677,6 +677,15 @@ func (r KubernetesFluxConfigurationResource) Update() sdk.ResourceFunc {
 			}
 
 			if metadata.ResourceData.HasChange("kustomizations") {
+				if _, exists := metadata.ResourceData.GetOk("git_repository"); exists {
+					_, configurationProtectedSettings, err := expandGitRepositoryDefinitionModel(model.GitRepository)
+					if err != nil {
+						return err
+					}
+					properties.Properties.ConfigurationProtectedSettings = configurationProtectedSettings
+				} else if _, exists = metadata.ResourceData.GetOk("bucket"); exists {
+					_, properties.Properties.ConfigurationProtectedSettings = expandBucketDefinitionModel(model.Bucket)
+				}
```

Comment:
This fixes the issue if `kustomizations` is updated, however if `continuous_reconciliation_enabled` were to be updated then users would hit the same bug since `ConfigurationProtectedSettings` was set to nil on line 644 and hasn't been repopulated.

I think we should pull this logic out and execute it after checking all the properties for changes and if `ConfigurationProtectedSettings` is still nil (so we don't unnecessarily process it again), so move this to line 695
```
if properties.Properties.ConfigurationProtectedSettings = nil {
// add logic here
}
```

Reply:
@stephybun, thank you very much for your answer.
Just pushed changes according to what you proposed.
Everything has been tested - working as expected.

Reply:
@stephybun, could you check my answer? 
Please, it's urgent for us.